pallets/aggregate/src/lib.rs

```diff
@@ -781,6 +832,62 @@ pub mod pallet {
 
             Ok(owner.post_info(None))
         }
+
+        /// Set gas price for a given domain ID
+        /// Gas price is X in the following formula - 1 TOKEN in destination chain = X origin tokens
+        #[pallet::call_index(4)]
+        pub fn set_gas_price(
+            origin: OriginFor<T>,
+            domain_id: u32,
+            gas_price: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
+            let owner = User::<T::AccountId>::from_origin::<T>(origin)?;
+
+            Domains::<T>::try_mutate(domain_id, |domain| -> Result<(), DispatchError> {
+                let domain = domain.as_mut().ok_or(Error::<T>::UnknownDomainId)?;
+
+                if !owner.can_handle_domain::<T>(domain) {
+                    return Err(BadOrigin.into());
+                }
+
+                domain.destination.gas_price = gas_price;
+                Ok(())
+            })?;
+
+            Ok(owner.post_info(None))
+        }
+
+        /// Hold (lock) tokens to pay dispatch fees
+        /// Only the domain owner can call this extrinsic.
+        #[pallet::call_index(5)]
+        pub fn hold_tokens_dispatch_fee(
+            origin: OriginFor<T>,
+            domain_id: u32,
+            amount: BalanceOf<T>,
+        ) -> DispatchResultWithPostInfo {
```

Comment:
Also here DispatchResult is enough

Reply:
does not apply anymore, this extrinsic will me removed